nix/module/worker.nix

```
@@ -0,0 +1,116 @@
+{ pkgs, lib, config, ... }:
+
+{
+  options.services.sioworker = {
+    enable = lib.mkEnableOption "sioworker";
+
+    package = lib.mkPackageOption pkgs "sioworkers" {
+      default = [ "python38Packages" "sioworkers" ];
+    };
+
+    filetrackerUrl = lib.mkOption {
+      default =
+        if ((builtins.hasAttr "filetracker" config.services) && config.services.filetracker.enable) then
+          "http://${config.services.filetracker.listenAddress}:${builtins.toString config.services.filetracker.port}" else null;
+      description = "The filetracker URL for the sioworker to connect to";
+      type = lib.types.str;
+    };
+
+    filetrackerCache = lib.mkOption {
+      default = "/var/cache/sioworker/filetracker";
+      description = lib.mdDoc ''
+        The path of the filetracker cache for this sioworker
+
+        IMPORTANT: The sioworker service **needs to have access to this location** and this is **not** handled automatically!!
+        **You** have to ensure that the sioworker systemd service will execute with the appriopriate permissions **yourself**.
+      '';
+      type = lib.types.path;
+    };
+
+    sioworkersd = lib.mkOption {
+      default = { };
+      description = "The sioworkersd for the sioworker to connect to";
+      type = lib.types.submodule {
+        options.host = lib.mkOption {
+          default = "localhost";
+          description = "The sioworkerd host";
+          type = lib.types.str;
+        };
+        options.port = lib.mkOption {
+          default = config.services.sioworkersd.worker.port;
+          description = "The sioworkerd port";
+          type = lib.types.port;
+        };
+      };
+    };
+
+    concurrency = lib.mkOption {
+      default = "auto";
+      description = "The amount of threads the sioworker should use";
+      type = with lib.types; oneOf [ (strMatching "auto") ints.positive ];
+    };
+
+    memoryLimit = lib.mkOption {
+      default = 1024;
+      description = "The amount of memory (in megabytes) this worker will use";
+      type = lib.types.ints.unsigned;
+    };
+  };
+
+  config =
+    let
+      cfg = config.services.sioworker;
+      python = cfg.package.pythonModule;
+      autoFiletrackerCache = lib.hasPrefix "/var/cache/sioworker" cfg.filetrackerCache;
```

This is a pretty hacky solution, maybe it should be replaced by something like this:
```suggestion
      ownedFiletrackerCache = cfg.filetrackerCache != null;
      filetrackerCache = if cfg.filetrackerCache == null then "/var/cache/sioworker/filetracker" else cfg.filetrackerCache;
```
and then we could make `cfg.filetrackerCache` default to null.